xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/api/Object.java

```diff
@@ -102,6 +104,15 @@ public boolean equals(java.lang.Object arg0)
         Object o = (Object) arg0;
         return o.getXWikiContext().equals(getXWikiContext()) && this.element.equals(o.element);
     }
+    
+    @Override
+    public int hashCode()
+    {
+        return new HashCodeBuilder(3, 17)
+            .append(getXWikiContext())
+            .append(element)
+            .toHashCode();
+    }
```

Comment:
This doesn't look correct. For ex I wouldn't put the xwiki context in it.

Also please provide some test cases when you modify equals or hashcode to prove that the changes work. Thanks!

Reply:
Akshit please forget all the classes in com.xpn.xwiki.api package and any other weird equals that includes test on the context...